statuspage/src/main/java/com/jagrosh/jdautilities/statuspage/endpoints/Incidents.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2016-2018 John Grosh (jagrosh) & Kaidan Gustave (TheMonitorLizard)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.jagrosh.jdautilities.statuspage.endpoints;
+
+import com.jagrosh.jdautilities.statuspage.data.Incident;
+import com.jagrosh.jdautilities.statuspage.data.Page;
+
+import javax.annotation.Nonnull;
+import javax.annotation.concurrent.Immutable;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+import java.util.ListIterator;
+
+@Immutable
+public class Incidents implements List<Incident>
+{
+    @Nonnull
+    protected final Page page;
+    @Nonnull
+    protected final List<Incident> incidents;
+
+    public Incidents(@Nonnull Page page, @Nonnull List<Incident> incidents)
+    {
+        this.page = page;
+        this.incidents = incidents;
+    }
+
+    @Nonnull
+    public Page getPage()
+    {
+        return page;
+    }
+
+    @Nonnull
+    public List<Incident> getIncidents()
+    {
+        return incidents;
+    }
+
+    @Override
+    public int size() {return incidents.size();}
+
+    @Override
+    public boolean isEmpty() {return incidents.isEmpty();}
+
+    @Override
+    public boolean contains(Object o) {return incidents.contains(o);}
+
+    @Override
+    @Nonnull
+    public Iterator<Incident> iterator() {return incidents.iterator();}
+
+    @Override
+    @Nonnull
+    public Object[] toArray() {return incidents.toArray();}
+
+    @Override
+    @Nonnull
+    @SuppressWarnings("SuspiciousToArrayCall")
+    public <T> T[] toArray(@Nonnull T[] a) {return incidents.toArray(a);}
+
+    @Override
+    public boolean add(Incident incident) {return incidents.add(incident);}
```

Comment:
This class is marked as `@Immutable` yet we are allowing modification of the underlying list?
This should throw `UnsupportedOperationException` or the `List` interface should not be implemented at all.
(This goes for all other methods that might modify this object)